subkey/src/main.rs

```diff
@@ -16,31 +22,50 @@
 
 #![cfg_attr(feature = "bench", feature(test))]
 #[cfg(feature = "bench")]
+
 extern crate test;
 extern crate substrate_primitives;
 extern crate rand;
+extern crate num_cpus;
 
 #[macro_use]
 extern crate clap;
+extern crate pbr;
 
+use vanity::OutputFormat;
+use keyspecs::KeySpecs;
 use substrate_primitives::{ed25519::Pair, hexdisplay::HexDisplay};
 
 mod vanity;
+mod keypair;
+mod keyspecs;
 
 fn main() {
 	let yaml = load_yaml!("cli.yml");
 	let matches = clap::App::from_yaml(yaml).get_matches();
 
 	match matches.subcommand() {
 		("vanity", Some(matches)) => {
-			let desired: String = matches.value_of("pattern").map(str::to_string).unwrap_or_default();
-			let key = vanity::generate_key(&desired).expect("Key generation failed");
-			println!("Seed {} (hex: 0x{}) - {} ({}%)",
-				key.pair.public().to_ss58check(),
-				HexDisplay::from(&key.pair.public().0),
-				HexDisplay::from(&key.seed),
-				key.score);
+			let number: u32 = matches.value_of("number").map(str::to_string).unwrap_or_default().parse::<u32>().unwrap();
+			let threads: u8 = matches.value_of("threads").map(str::to_string).unwrap_or_default().parse::<u8>().unwrap();
+			let format: OutputFormat = match matches.value_of("format").unwrap_or_default() {
+				"csv" => OutputFormat::Csv,
+				_ => OutputFormat::Stdout,
+			};
+			let key_specs = KeySpecs {
+				 desired_pattern: matches.value_of("pattern").map(str::to_string).unwrap_or_default(),
+				 case_sensitive: matches.is_present("case_sensitive"),
+				 paranoiac: matches.is_present("paranoiac"),
+				 minscore: match matches.value_of("minscore").map(str::to_string).unwrap_or_default().parse::<u8>().unwrap() {
+					 m @ 0...100 => m as f32,
+					_ => 100_f32,
```

Comment:
perhaps this could be more efficiently written as `::std::cmp::min(m, 100) as f32`?

Reply:
ok, fixed.